Fix W18-51 to -65 abstracts (fixes #714)

GuyAglionby opened the pull request:
Semi-automatically done -- used [this script](https://gist.github.com/GuyAglionby/a9d184ddec0253fe8d49c49acd88bbe5) then manually verified the changes (though there were a lot so it's possible I missed a couple of extraction errors). 

It looks like there were two classes of changes: human-made (from minor edits to more serious reworks) and errors (e.g. text cut off mid sentence). Hope this isn't too large a change!

davidweichiang commented:
Thanks! I've never used pdfquery; how does it know where to stop extracting lines for the abstract?

davidweichiang commented on data/xml/W18.xml:
```diff
@@ -12115,6 +12186,7 @@
       <author><first>Alexander</first><last>Fraser</last></author>
       <pages>648–654</pages>
       <url>W18-6446</url>
+      <abstract>We present the LMU Munich machine translation systems for the English–German language pair. We have built neural machine translation systems for both translation directions (English→German and German→English) and for two different domains (the biomedical domain and the news domain). The systems were used for our participation in the WMT18 biomedical translation task and in the sha ...
```
delete `1,2`

davidweichiang commented on data/xml/W18.xml:
```diff
@@ -12463,6 +12572,7 @@
       <author><first>Stelios</first><last>Piperidis</last></author>
       <pages>928–933</pages>
       <url>W18-6484</url>
+      <abstract>This paper describes the submission of the Institute for Language and Speech Processing/Athena Research and Innovation Center (ILSP/ARC) for the WMT 2018 Parallel Corpus Filtering shared task. We explore several properties of sentences and sentence pairs that our system explored in the context of the task with the purpose of clustering sentence pairs according to their appropriate ...
```
T -> MT

davidweichiang commented on data/xml/W18.xml:
```diff
@@ -12905,7 +13020,7 @@
       <pages>296–306</pages>
       <url>W18-6536</url>
-      <abstract>Question Generation is the task of automatically creating questions</abstract>
+      <abstract>Question Generation is the task of automatically creating questions from textual input. In this work we present a new Attentional Encoder–Decoder Recurrent Neural Network model for automatic question generation. Our model incorporates linguistic features and an additional sentence embedding to capture meaning at both sentence and word levels. The linguistic features are designed t ...
```
`Bleu 4` -> `Bleu_4`

GuyAglionby commented:
Thanks for the review and taking the time to find examples of errors! I have rewritten the scraping code to be both able to extract formatting and more robust in retrieving all characters in the document. The resulting changes are mostly for formatting, but also some short phrases that the original code missed.

I think most of the errors you noticed were because of slight misalignment on the y axis of some characters, which led to pdfquery misordering or ignoring them. I've rewritten this code to deal with a wider range of edge cases ([here](https://gist.github.com/GuyAglionby/4b55d00803710f2e2e9877fd18b5a491) if interested).

To your first question, it turns out that the abstract itself is all contained within a single xml element, so I just iterate through all of the characters in that ([this xpath](https://gist.github.com/GuyAglionby/4b55d00803710f2e2e9877fd18b5a491#file-fix-abstracts-py-L224)). For some papers the first header is included in that element; this is "Introduction" 99% of the time so I filter that out.

davidweichiang commented:
Thank you! I can’t get this until mid-January, but maybe someone else will be able to.

GuyAglionby commented:
Thanks! Will do
